[Enhancement] Change some default session variables for MV refresh

LiShuMing opened the pull request:
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

LiShuMing committed:
[Refactor] Change olap_table_query_rewrite_consistency to query_rewrite_consistency as materialized view's property (#28328)

- Change olap_table_query_rewrite_consistency to
query_rewrite_consistency to make it a property as
MV.`query_rewrite_consistency` can be :
     - disable: this mv cannot be used for rewrite
- checked: this mv can only be used for rewrite only this mv is
satisfied for its staleness
     - loose: this mv will be always used for rewrite.
- To be compatible with old property
`force_external_table_query_rewrite`,
`force_external_table_query_rewrite` is still kept but will be
Deprecated later.
- Refactor getMVToRefreshPartitionNames into two methods:
getNonPartitionedMVRefreshPartitions and
getPartitionedMVRefreshPartitions
- Change logPrepare/logRewrite into log format .、

Signed-off-by: Kevin Li <ming.moriarty@gmail.com>

LiShuMing committed:
[BugFix] Materialized view should update all its refreshed base table partitions after refresh (#28093)

Changes:
- After mv refreshed, SR only updated base table partitions from plan
which may be partition pruned.
This is not expected, I changed it to update its base table partitions
if the mv has updated.
- For external ref table, add `externalRefBaseTableMVPartitionMap ` in
taskRunContext;
- Add `refBaseePartitionsToRefreshMap ` and `basePartitionsToRefreshMap
` in task_runs information_schema.

Signed-off-by: Kevin Li <ming.moriarty@gmail.com>

LiShuMing committed:
[Enhancement] Change some default session variables for MV refresh (#28482)

Signed-off-by: Kevin Li <ming.moriarty@gmail.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 332 / 418 (79.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|8|13|61.54%|[376, 377, 378, 391, 392]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|4|6|66.67%|[596, 603]|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptimizerTraceUtil.java|2|3|66.67%|[149]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|153|205|74.63%|[220, 221, 225, 226, 227, 265, 428, 508, 582, 583, 584, 585, 586, 588, 596, 597, 598, 599, 600, 603, 616, 617, 618, 683, 685, 686, 688, 896, 897, 907, 919, 925, 979, 988, 1010, 1206, 1207, 1208, 1211, 1212, 1215, 1216, 1217, 1218, 1219, 1220, 1225, 1231, 1235, 1241, 1242, 1243]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|39|48|81.25%|[266, 267, 268, 396, 397, 400, 406, 407, 420]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|75|89|84.27%|[593, 676, 1220, 1225, 1233, 1245, 1256, 1263, 1264, 1275, 1304, 1323, 1325, 1326]|
|:large_blue_circle:|com/starrocks/scheduler/persist/MVTaskRunExtraMessage.java|7|8|87.50%|[98]|
|:large_blue_circle:|com/starrocks/sql/optimizer/MvRewritePreprocessor.java|9|10|90.00%|[200]|
|:large_blue_circle:|com/starrocks/scheduler/MvTaskRunContext.java|12|13|92.31%|[116]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OptExpressionDuplicator.java|1|1|100.00%|[]|

LiShuMing committed:
[BugFix] Rethrow exception when refresh mv failed  (#28554)

1. Fix bugs introduced by
[22663](https://github.com/StarRocks/starrocks/pull/28093/files#)
  - Rethrow exception when refresh mv failed
  - Support cloud native table for update meta
3. Fix bugs introduced by
[22663](https://github.com/StarRocks/starrocks/pull/22663/)
  - revert the codes first.

Signed-off-by: Kevin Li <ming.moriarty@gmail.com>
